tests/integration_tests/test_hyperopt.py

```diff
@@ -562,3 +491,94 @@ def test_hyperopt_old_config(csv_filename, tmpdir, ray_cluster):
     rel_path = generate_data(input_features, output_features, csv_filename)
 
     hyperopt(old_config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
+
+
+@pytest.mark.distributed
+def test_hyperopt_nested_parameters(csv_filename, tmpdir, ray_cluster):
+    config = {
+        INPUT_FEATURES: [
+            {"name": "cat1", TYPE: "category", "encoder": {"vocab_size": 2}},
+            {"name": "num1", TYPE: "number"},
+        ],
+        OUTPUT_FEATURES: [
+            {"name": "bin1", TYPE: "binary"},
+        ],
+        TRAINER: {"epochs": 2},
+        HYPEROPT: {
+            EXECUTOR: {
+                TYPE: "ray",
+                "time_budget_s": 200,
+                "cpu_resources_per_trial": 1,
+                "num_samples": 4,
+                "scheduler": {TYPE: "fifo"},
+            },
+            "search_alg": {TYPE: "variant_generator"},
+            "parameters": {
+                ".": {
+                    "space": "choice",
+                    "categories": [
```

Comment:
So we're essentially hyperopting over the combo of a `combiner` with type `tabnet` and the `adam` optimizer OR a `combiner` with type `concat` and the default trainer with specific type of learning rate scaling?

Reply:
Correct.